src/webapi/xml_http_request.rs

```diff
@@ -0,0 +1,129 @@
+use webapi::event_target::{IEventTarget, EventTarget};
+use webcore::value::{
+    Reference,
+    Value,
+};
+use webcore::try_from::TryInto;
+
+/// XMLHttpRequest object to make http requests
+/// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest)
+pub struct XMLHttpRequest( Reference );
+
+reference_boilerplate! {
+    XMLHttpRequest,
+    instanceof XMLHttpRequest
+    convertible to EventTarget
+}
+
+/// A number indicating the state of the `XMLHttpRequest`.
+///
+/// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/readyState)
+#[derive(Copy, Clone, PartialEq, Eq, Debug)]
+pub enum ReadyState {
+    /// Client has been created. open() not called yet.
+    Unsent,
+    /// open() has been called.
+    Opened,
+    /// send() has been called, and headers and status are available.
+    HeadersReceived,
+    /// Downloading; responseText holds partial data.
+    Loading,
+    /// The operation is complete.
+    Done,
+}
+
+impl IEventTarget for XMLHttpRequest {}
+
+
+impl XMLHttpRequest {
+    /// Create new XMLHttpRequest
```

Comment:
Minor grammar nitpick (this also applies to other doc comments): 'Create' -> 'Creates', and also a dot at the end and put identifiers (like `XMLHttpRequest`) inside backticks. (: